### PR TITLE
Upgrade RESTHeart

### PR DESCRIPTION
**Pull request by ujibang**
now that you support java 17 runtime, it time to upgrade RESTHeart!

### CODE REVIEW

File: java/restheart/config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 framework:
   website: restheart.org
-  version: 5.4
+  version: 7.0.0-SNAPSHOT
```
Comment (waghanza) [inline java/restheart/config.yaml:3]:
```suggestion
  version: 7.0
```

File: java/restheart/pom.xml [hunk 2]
```diff
@@ -7,10 +7,9 @@
   <packaging>jar</packaging>
 
   <properties>
-    <mongodb.version>4.2</mongodb.version>
     <maven.compiler.source>17</maven.compiler.source>
     <maven.compiler.target>17</maven.compiler.target>
-    <restheart.version>[5.4,5.5)</restheart.version>
+    <restheart.version>[6.0,7.0)</restheart.version>
```
Comment (waghanza) [inline java/restheart/pom.xml:12]:
```suggestion
    <restheart.version>[6.0,6.1)</restheart.version>
```

**Review by waghanza — CHANGES_REQUESTED** [review]
Thank for your contribution 🎉 

Wonder why `renovatebot` does not trigger any update PR 🤔 

Some changes ... and we are good to go

File: java/restheart/config.yaml (hunk 1 above)
Comment (ujibang) [reply java/restheart/config.yaml:3]: Latest release is 6.5.0

File: java/restheart/pom.xml (hunk 2 above)
Comment (ujibang) [reply java/restheart/pom.xml:12]: [6.0,7.0) also selects 7.0.0-SNAPSHOT. Although v7 has not yet been released, it is stable.
If you want to limit the test to stable releases, please use: `<restheart.version>[6.0,6.100)</restheart.version>`

Comment (waghanza) [reply java/restheart/pom.xml:12]: I prefer to have the last stable and published released. `renovate` should ensure updates

Comment (waghanza) [reply java/restheart/pom.xml:12]: Maybe
```suggestion
    <restheart.version>[6.5,6.6)</restheart.version>
```

File: java/restheart/config.yaml (hunk 1 above)
Comment (waghanza) [reply java/restheart/config.yaml:3]: Then

```suggestion
  version: 6.5
```

**Pull request closed without merging**